src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -479,9 +479,7 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 							break;
 
 						default:
-							Debug.Fail ("Unsupported member type");
-							reflectionContext.RecordUnrecognizedPattern ($"Reflection call '{calledMethod.FullName}' inside '{callingMethodBody.Method.FullName}' is of unexpected member type.");
-							break;
+							throw new InternalErrorException ($"Reflection call '{calledMethod.FullName}' inside '{callingMethodBody.Method.FullName}' is of unexpected member type.");
```

Comment:
I don't think this should be fatal, probably not even error

Reply:
Because of how this switch-case is nested, it's an invariant that the `IntrinsicId` will be matched. In other words, if we ever hit this line is because something really weird is going on.

Reply:
Agreed - this is basically assert, but based on other discussions we've switched all asserts to internal errors - which are fatal.